compiler/transforms/insert_acc_op.py

```diff
@@ -0,0 +1,63 @@
+from dataclasses import dataclass
+
+from xdsl.dialects import builtin
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.traits import SymbolTable
+
+from compiler.accelerators.registry import AcceleratorRegistry
+from compiler.dialects import acc
+
+
+@dataclass
+class InsertAcceleratorOpPattern(RewritePattern):
+    acc_op: acc.AcceleratorOp
+    """
+    Pattern that attaches the AcceleratorOp specified by acc_op to
+    the module symbol table if the symbol wasn't there already.
+    Raises a RuntimeError if the symbol is already present.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: builtin.ModuleOp, rewriter: PatternRewriter):
+        # Get module symbol table and attach accelerator op to it
+        t = op.get_trait(SymbolTable)
+        assert t is not None
+        acc_string = self.acc_op.name_prop.string_value()
+        if t.lookup_symbol(op, acc_string) is not None:
+            raise RuntimeError(
+                f"Cannot insert op: @{acc_string}, already present in symbol table."
+            )
+        t.insert_or_update(op, self.acc_op)
```

Comment:
does adding something to the symbol table also put it in the IR?

Reply:
aha, now i see the test. nice

Reply:
Apparently it does hahaha